src/jetzig/views/Route.zig

```diff
@@ -19,33 +22,53 @@ pub const ViewType = union(Action) {
     delete: ViewWithId,
 };
 
+pub const StaticViewType = union(Action) {
+    index: StaticViewWithoutId,
+    get: StaticViewWithId,
+    post: StaticViewWithoutId,
+    put: StaticViewWithId,
+    patch: StaticViewWithId,
+    delete: StaticViewWithId,
+};
+
+pub const ViewType = union(enum) {
+    static: StaticViewType,
+    dynamic: DynamicViewType,
+};
+
 name: []const u8,
 action: Action,
-view: ViewType,
+uri_path: []const u8,
+view: ?ViewType = null,
+static_view: ?StaticViewType = null,
+static: bool,
 render: RenderFn = renderFn,
+renderStatic: RenderStaticFn = renderStaticFn,
+template: []const u8,
 
-pub fn templateName(self: Self, allocator: std.mem.Allocator) ![]const u8 {
-    if (std.mem.eql(u8, self.name, "app.views.index") and self.action == .index)
-        return try allocator.dupe(u8, "index");
-
-    const underscored_name = try std.mem.replaceOwned(u8, allocator, self.name, ".", "_");
-    defer allocator.free(underscored_name);
-
-    // FIXME: Store names in a normalised way so we don't need to do this stuff:
-    const unprefixed = try allocator.dupe(u8, underscored_name["app_views_".len..self.name.len]);
-    defer allocator.free(unprefixed);
-
-    const suffixed = try std.mem.concat(allocator, u8, &[_][]const u8{
-        unprefixed,
-        "_",
-        @tagName(self.action),
-    });
+fn renderFn(self: Self, request: *jetzig.http.Request) anyerror!jetzig.views.View {
```

Comment:
Don't use anyerror, use !jetzig.views.View

Reply:
I don't think this is avoidable - storing a function pointer without explicit `anyerror` fails:
```zig
const FunctionContainer = struct {
    function: *fn() !void,
};
```
```console
src/main.zig:4:21: error: function prototype may not have inferred error set
    function: *fn() !void,
```

Reply:
Oh, that's valid, didn't think of that, but this only resolves the render functions ones